frontend/wasm/src/component/lift_exports.rs

```diff
@@ -81,8 +166,106 @@ pub fn generate_export_lifting_function(
         .map(|ba| ba as ValueRef)
         .collect();
 
-    // NOTE: handle CC lifting/lowering for non-scalar types
-    // see https://github.com/0xMiden/compiler/issues/369
+    // Export lifting: The core exported function returns a pointer to the result
+    // We need to:
+    // 1. Load the data from that pointer into the "flattened" representation (primitive types)
+    // 2. Return it as individual values (tuple)
+
+    let exec = fb.exec(core_export_func_ref, core_export_func_sig, args, span)?;
+
+    let borrow = exec.borrow();
+    let results = borrow.results().all();
+
+    // The core function should return a single pointer (as i32)
+    assert_eq!(results.len(), 1, "expected single result");
+    let result_ptr = results.into_iter().next().unwrap().borrow().as_value_ref();
+
+    // Load values from the core function's result pointer using recursive loading
+    let mut return_values = SmallVec::<[ValueRef; 8]>::new();
+
+    // Load results using the recursive function from canon_abi_utils
+    assert_eq!(
+        export_func_ty.results.len(),
+        1,
+        "expected a single result in the component-level export function"
+    );
+    let result_type = &export_func_ty.results[0];
+
+    load(&mut fb, result_ptr, result_type, &mut return_values, span)?;
+
+    assert!(
```

Comment:
*Suggestion:* Same as my previous one regarding the information available to us if this assertion is raised

Reply:
Done in https://github.com/0xMiden/compiler/pull/533/commits/91310d5358ff12249f677c54c8234e9cc0a39dc3